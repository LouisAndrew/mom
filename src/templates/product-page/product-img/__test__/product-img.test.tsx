import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import ProductImg from '..';

describe('Product img component.', () => {
    const mockImgs = [
        {
            image:
                'https://res.cloudinary.com/dsvdffre0/image/upload/v1597137791/webaliser-_TPTXZd9mOo-unsplash_pqtm1r.jpg',
            imgAlt: 'Beautiful house',
        },
        {
            image:
                'https://res.cloudinary.com/dsvdffre0/image/upload/v1597137791/jacques-bopp-Hh18POSx5qk-unsplash_lweaht.jpg',
            imgAlt: 'Beautiful house 2',
        },
        {
            image:
                'https://res.cloudinary.com/dsvdffre0/image/upload/v1597137792/digital-marketing-agency-ntwrk-g39p1kDjvSY-unsplash_kh6gti.jpg',
            imgAlt: 'Beautiful house 3',
        },
    ];
    const el: React.ReactElement = <ProductImg imgs={mockImgs} />;

    afterEach(cleanup);

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(el, div);
    });

    it('should react to user input correctly', () => {
        const { getByTestId, getByAltText } = render(el);

        const mainImg: HTMLImageElement = getByTestId(
            'main'
        ) as HTMLImageElement;
        const img2: HTMLElement = getByAltText(mockImgs[2].imgAlt);
        fireEvent.click(img2);

        expect(mainImg.src).toBe(mockImgs[2].image);
    });

    it('matches snapshot', () => {
        const tree = renderer.create(el).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
