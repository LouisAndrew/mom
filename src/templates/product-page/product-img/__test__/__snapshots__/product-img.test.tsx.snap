// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product img component. matches snapshot 1`] = `
<div
  className="sc-AxirZ nQITg"
  width={
    Array [
      "fit-content",
    ]
  }
>
  <div
    className="sc-AxjAm kQoZeu"
    height={
      Array [
        180,
        222,
        300,
      ]
    }
    overflow="hidden"
    width={
      Array [
        270,
        330,
        450,
      ]
    }
  >
    <img
      alt="Image not found"
      data-testid="main"
      src="https://res.cloudinary.com/dsvdffre0/image/upload/v1597323293/sarah-kilian-52jRtc2S_VE-unsplash_rgrekp.jpg"
    />
  </div>
  <div
    className="sc-AxiKw ldVjDC"
    display="flex"
  >
    <div
      className="sc-AxhCb fZIOmp"
      height={
        Array [
          40,
          40,
          60,
        ]
      }
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={-3}
      width={
        Array [
          60,
          60,
          90,
        ]
      }
    >
      <img
        alt="Beautiful house"
        src="https://res.cloudinary.com/dsvdffre0/image/upload/v1597137791/webaliser-_TPTXZd9mOo-unsplash_pqtm1r.jpg"
      />
    </div>
    <div
      className="sc-AxhCb fZIOmp"
      height={
        Array [
          40,
          40,
          60,
        ]
      }
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={-2}
      width={
        Array [
          60,
          60,
          90,
        ]
      }
    >
      <img
        alt="Beautiful house 2"
        src="https://res.cloudinary.com/dsvdffre0/image/upload/v1597137791/jacques-bopp-Hh18POSx5qk-unsplash_lweaht.jpg"
      />
    </div>
    <div
      className="sc-AxhCb fZIOmp"
      height={
        Array [
          40,
          40,
          60,
        ]
      }
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={-1}
      width={
        Array [
          60,
          60,
          90,
        ]
      }
    >
      <img
        alt="Beautiful house 3"
        src="https://res.cloudinary.com/dsvdffre0/image/upload/v1597137792/digital-marketing-agency-ntwrk-g39p1kDjvSY-unsplash_kh6gti.jpg"
      />
    </div>
  </div>
</div>
`;
